common/src/main/java/io/crate/types/DataTypes.java

```diff
@@ -333,17 +340,30 @@ public static DataType ofName(String name) {
         .put("object", ObjectType.untyped())
         .put("nested", ObjectType.untyped()).build();
 
+    private static final Map<Integer, String> TYPE_IDS_TO_MAPPINGS = Map.ofEntries(
+        entry(TIMESTAMP.id(), "date"),
+        entry(STRING.id(), "text"),
+        entry(BYTE.id(), "byte"),
+        entry(BOOLEAN.id(), "boolean"),
+        entry(IP.id(), "ip"),
+        entry(DOUBLE.id(), "double"),
+        entry(FLOAT.id(), "float"),
+        entry(SHORT.id(), "short"),
+        entry(INTEGER.id(), "integer"),
+        entry(LONG.id(), "long"),
+        entry(ObjectType.ID, "object"),
+        entry(GEO_SHAPE.id(), "geo_shape"),
+        entry(GEO_POINT.id(), "geo_point")
+    );
+
     @Nullable
-    public static DataType ofMappingName(String name) {
-        return MAPPING_NAMES_TO_TYPES.get(name);
+    public static String esMappingNameFrom(int typeId) {
```

Comment:
what about adding a e.g. `mapperName()` method to `DataType` instead?

Reply:
i thought about it, but it would also touch some unrelated `DataType` child classes like in aggregations, e.g. `AverageStateType`, collection types, etc. We can, of course, have have a default implementation if `mapperName` that returns probably null, but is it really worth to change the `DataType` interface in this case. It does not enforce an implementation of `mapperName`.